[MIG][17.0] sale_automatic_workflow_payment_mode: Migration to 17.0

john-herholz-dt opened the pull request:
Simple migration, no changes.

Commit:
[WIP][9.0] Add sale_automatic_workflow_payment_mode (#297)

[ADD] sale_automatic_workflow_payment_mode

pedrobaeza committed:
[MIG] Make modules uninstallable

[MIG] Rename manifest files

lmignon committed:
[MIG] sale_automatic_workflow_payment_mode: Upgrade to v10

Commit:
[IMP] sale_automatic_workflow_payment_mode: improve test coverage

lmignon committed:
[IMP] sale_automatic_workflow_payment_mode: OCA conventions

[IMP] sale_automatic_workflow_payment_mode: Use related fields for payment_filter_domain & char_domain widget

hugosantosred committed:
[MIG] sale_automatic_workflow_payment_mode to V11

murtuzasaleh committed:
[11.0] [MIG]sale_procurement_group_by_line: Migration to 11.0.

florian-dacosta committed:
Migration to version 12.0

sbidoul committed:
[IMP] sale_automatic_workflow_payment_mode cosmetics

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 20.0% (2 of 10 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow_payment_mode/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 30.0% (3 of 10 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow_payment_mode/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 40.0% (4 of 10 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow_payment_mode/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (10 of 10 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow_payment_mode/zh_CN/

Galoperd committed:
Translated using Weblate (Spanish)

Currently translated at 90.0% (9 of 10 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow_payment_mode/es/

Commit:
Add sale_automatic_workflow_job

Use Queue Jobs to process the Sales Automatic Workflow actions.

The default behavior of the automatic workflow module is to use a
scheduled action that searches all the record that need a workflow
action and sequentially process all of them.

It can hit some limits when the number of records is too high.

This module keeps the scheduled action to search the records, but
instead of directly executing the actions (confirm a sales order,
create invoices for a sales order, validate invoices, ...), it
creates one job per operation to do.

It uses an identity key on the jobs so it will not create the same
job for the same record and same operation twice.

~

I needed to extract methods in `sale_automatic_workflow` in order
to be able to execute them as jobs.

A new decorator "job_auto_delay" (to eventually move in queue_job) makes
these methods automatically delayed when called.

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 30.0% (3 of 10 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow_payment_mode/sl/

renatonlima committed:
[FIX] pep8 and pylint

ThomasBinsfeld committed:
[MIG] sale_automatic_workflow_payment_mode from 12.0 to 13.0

ThomasBinsfeld committed:
[MIG] sale_automatic_workflow_payment_mode: run pre-commit

hailangvn committed:
[IMP] sale_automatic_workflow_payment_mode: black, isort, prettier

hailangvn committed:
[MIG] sale_automatic_workflow_payment_mode: Migration to 14.0

hailangvn committed:
[IMP] sale_automatic_workflow_payment_mode: sync SO workflow process, increase coverage

victoralmau committed:
[IMP] sale_automatic_workflow:
- Move register_payment option from sale_automatic_workflow_payment_mode to
sale_automatic_workflow.
- Add _register_payments function in sale_automatic_workflow.
TT31208

paulius-sladkevicius committed:
[FIX] sale_automatic_workflow_mode: missing super()

Commit:
[FIX] sale_automatic_workflow_payment_mode: not register...

...payment when no fixed journal and remove sale_workflow_process

jcoux committed:
[14.0][FIX] Fix sale_automatic_workflow_payment_mode migration

Parts of `sale_automatic_workflow_payment_mode` have been merged into `sale_automatic_workflow`.
But these parts have been not removed from `sale_automatic_workflow_payment_mode`.

This PR clean both modules after this merge.

Originals PR of the merge: https://github.com/OCA/sale-workflow/pull/1789

ivantodorovich committed:
[FIX] sale_automatic_workflow_payment_mode: avoid overwritting/removing SO's workflow if payment.mode hasn't one set

It seems this feature was added during the 14.0 migration PR, here: 111bb97c7
But it wasn't like this before, and IMO it was ok.

OCA-git-bot committed:
sale_automatic_workflow_payment_mode 14.0.1.0.1

flachica committed:
[MIG] sale_automatic_workflow_payment_mode 15.0 &

    * Fix tests errors

florian-dacosta committed:
[16.0][MIG] Migrate sale_automatic_workflow_payment_mode to v16

Commit:
[UPD] Update sale_automatic_workflow_payment_mode.pot

OCA-git-bot committed:
[UPD] README.rst

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/es/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/pt_BR/

Commit:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/it/

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/it/

rousseldenis commented:
/ocabot migration sale_automatic_workflow_payment_mode

@john-herholz-dt Thanks for this. Don't forget to add dependency PR in this PR's description to help reviewers.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/it/

rousseldenis commented:
@john-herholz-dt Could you rebase ?

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (4 of 4 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_payment_mode/hr/

john-herholz-dt commented:
@rousseldenis I did a rebase. The check failure does not seem to be related to this PR.

john-herholz-dt commented:
@rousseldenis last time I said "check failure has nothing to do with this PR". I checked again and sure the failure is one of the tests here.
But I don't understand it.
I manually checked the behaviour on runboat and everything worked.

Any suggestion how to solve the test?

rousseldenis commented:
@john-herholz-dt The error is related to this PR.

See : https://github.com/OCA/sale-workflow/actions/runs/10438425127/job/28905725687?pr=2892#step:8:380

The 'done' state has been removed from Odoo core. The equivalent is `[("state", "=", "sale"), ("locked", "=", True)]`

john-herholz-dt commented:
Hello @rousseldenis ,

Sorry I should have given more context: It is the stock.picking record which seems to not be validated on the test_full_automatic workflow.

I checked it manually with the runbot and the picking was done.

```
  File "/__w/sale-workflow/sale-workflow/sale_automatic_workflow_payment_mode/tests/test_automatic_workflow_payment_mode.py", line 100, in test_full_automatic
    self.assertEqual(picking.state, "done")
AssertionError: 'assigned' != 'done'
```

rousseldenis commented:
Oops mixed models.

Maybe a module of this repo changed the behavior

john-herholz-dt commented:
@rousseldenis I was working again on it and tests are passing.
The module really did have some check for pickings, which failed previously.
I removed the check as it is not part of this module and there is also now a separate module sale_automatic_workflow_stock

rousseldenis commented on sale_automatic_workflow_payment_mode/tests/test_automatic_workflow_payment_mode.py:
```diff
@@ -96,5 +96,3 @@ def test_full_automatic(self):
         self.env["automatic.workflow.job"].run()
         self.assertEqual(invoice.payment_state, "paid")
-        picking = sale.picking_ids
-        self.assertEqual(picking.state, "done")
```
@john-herholz-dt Why removing this test?

john-herholz-dt replied:
@rousseldenis 
It was seperated in 17.0
sale_automatic_workflow is no longer validating the picking.
There is a new module sale_automatic_workflow_stock

Also, this module should not be responsible for checking something in the pickings - as it has no logic (code) for this.
Or am I wrong?

rousseldenis commented:
@john-herholz-dt Could you move migration changes in migration commit (and not in pre-commit one)? Thanks

john-herholz-dt committed:
[IMP] sale_automatic_workflow_payment_mode: pre-commit auto fixes

john-herholz-dt committed:
[MIG] sale_automatic_workflow_payment_mode: Migration to 17.0

john-herholz-dt commented:
@rousseldenis Is it better now?

If so, can you help me a bit please:
When I migrate things and do the pre-commit stuff, it will tell me wrong manifest version if I keep it.
Nevertheless the manifest change should be in the migration commit, if I understand you correctly.

What am I missing here, that I keep the error in the pre-commit step?

rousseldenis commented:
> @rousseldenis Is it better now?
> 
> If so, can you help me a bit please: When I migrate things and do the pre-commit stuff, it will tell me wrong manifest version if I keep it. Nevertheless the manifest change should be in the migration commit, if I understand you correctly.
> 
> What am I missing here, that I keep the error in the pre-commit step?

That's because you forgot `--no-verify`option in the procedure:

![image](https://github.com/user-attachments/assets/95a10752-1cb0-4e56-ae51-e24b1ba49daa)

john-herholz-dt commented:
I get the error two lines above in the documentation at: pre-commit run -a

but anyway, the PR should be fine now, isn't it?

rousseldenis commented on sale_automatic_workflow_payment_mode/tests/test_automatic_workflow_payment_mode.py:
```diff
@@ -25,7 +25,7 @@ def setUpClass(cls):
 
     def setUp(self):
-        super(TestAutomaticWorkflowPaymentMode, self).setUp()
+        super().setUp()
```
@john-herholz-dt Could you put that change in migration commit too ? Thanks

flotho commented:
Hi @john-herholz-dt , any chance to rebase ?

rousseldenis commented:
Closing this as very old. Please feel free to reopen it if needed
